### PR TITLE
fix(competition): Rate short courses instead of refusing them

### LINKED ISSUE #206: Pitch & putt courses fall back to the raw Handicap Index
## What happens

`TeeRating` validates `66 <= par <= 76` (`competition/domain/services/playing_handicap_calculator.py`). A federated pitch & putt is par ~54, so **every one of its tees raises `ValueError`**, gets skipped by `StrokeContextBuilder`, and all participants fall back to playing off their raw Handicap Index instead of a Course Handicap.

The same bound rejects `EXECUTIVE` courses.

## Why it matters

It is silent. The match is playable and the numbers look plausible, they are just not the WHS ones. A player off 18 gets 18 strokes instead of the ~0 that a par-54 course off SR 47 / CR 46.8 would give them — the allocation is wrong by the whole handicap.

It also splits the two implementations in a way we deliberately closed elsewhere: the frontend `Tee` value object accepts CR 45-90 and SR 40-160 precisely so these courses can be stored, so `MatchPlayStrokeAllocator` had to mirror the *backend's* narrower bounds to stay in parity (`RATED_MIN_*` constants). Widening one side without the other reopens the drift.

## Why it was not fixed with the rest

`TeeRating` is shared with `competition`. Widening it changes the playing handicaps of competitions that already exist, and their matches are generated once and stored — so this needs its own change with a decision about existing data, not a line in a bugfix branch.

## Suggested approach

1. Decide the real WHS bounds for short courses (pitch & putt, executive) and widen `MIN_PAR` / `MAX_PAR`, `MIN_COURSE_RATING`, `MIN_SLOPE_RATING` accordingly.
2. Work out whether already-generated competition matches need regenerating, or whether the change applies only to new ones.
3. Relax the mirrored `RATED_*` constants in `MatchPlayStrokeAllocator` (RyderCupWeb) to match, and regenerate `scripts/generate_parity_fixtures.py` so the parity test covers a short course.

## Where to look

- `src/modules/competition/domain/services/playing_handicap_calculator.py` — `TeeRating.__post_init__`
- `src/modules/quick_match/application/services/stroke_context_builder.py` — the `except ValueError` that skips the tee (it logs a warning once per course, so production logs will show which courses are affected)
- `RyderCupWeb/src/domain/services/MatchPlayStrokeAllocator.js` — `isRatable`

Found while reviewing the quick match stroke allocation branch (`fix/quick-match-handicap-scoring`).

### LINKED ISSUE #219: fix(competition): Generating matches 500s on any course that has a short tee
> **Correction to the original report:** I first wrote that this needs a player enrolled on the offending tee. It does not — see "The trigger is the course" below. The fix belongs in #206; this issue records the competition-side symptom.

Generating a round's matches returns **500** on any course that has at least one tee whose Course Rating or Slope Rating falls outside the range `TeeRating` validates.

## Reproduction

Son Parc - Par 71. Its ORANGE tee is a short course: par 58, CR **54.9**, and `TeeRating` requires 55–85.

```
POST /api/v1/competitions/rounds/{round_id}/matches/generate
-> 500 Internal Server Error
```

```
File "generate_matches_use_case.py", line 261, in _build_handicap_data
    context = TeeContextBuilder.build(golf_course)
File "tee_context_builder.py", line 73, in build
    tee_ratings[tee_key] = TeeContextBuilder._rating_for(tee, course_par, golf_course)
File "tee_context_builder.py", line 107, in _rating_for
    return TeeRating(
File "playing_handicap_calculator.py", line 51, in __post_init__
    raise ValueError(
ValueError: course_rating must be between 55.0 and 85.0, got 54.9
```

The round is left with no matches and no way to generate them.

## The trigger is the course, not the tee anyone plays

`TeeContextBuilder.build` rates **every** tee of the course:

```python
for tee in golf_course.tees:
    tee_ratings[tee_key] = TeeContextBuilder._rating_for(tee, course_par, golf_course)
```

Verified by re-running the generation with all four players enrolled on YELLOW, a valid tee: still 500. One unratable tee anywhere in the course is enough, even if nobody is on it.

`reassign_match_players_use_case` builds the same context, so it fails the same way. The scoring view does not — it goes through `hole_card_for`, which does not rate anything.

## Why the existing fallback does not catch it

`_rating_for` anticipates an unusable tee:

> Si ese par no sirve —se sale del rango WHS, o la tarjeta viene malformada— se cae al del campo, **en vez de dejar la ronda sin poder generar partidos**.

But the retry inside `except` rebuilds `TeeRating` with the **same** `course_rating` and `slope_rating`, changing only the par. When the value out of range is the rating rather than the par, the fallback raises exactly like the first attempt — line 107 in the traceback is that second call. Whatever #206 lands on, this retry should not be able to raise the same error twice.

## Scope

**468 tees across 227 courses**, from the versioned RFEG dataset — not local seed data:

```sql
SELECT count(*) FROM golf_course_tees WHERE course_rating < 55 OR course_rating > 85;   -- 468
SELECT count(DISTINCT golf_course_id) FROM golf_course_tees WHERE course_rating < 55 OR course_rating > 85;  -- 227
```

Those 227 courses cannot host a competition at all today.

Fix tracked in #206, where the bounds themselves are being reconsidered — validating by course type, as `GolfCourse` already does, rather than one flat range.

### PR DESCRIPTION
**Pull request by agustinEDev**
Found while setting up a competition on the local cluster to test the per-tee par work in RyderCupWeb#424.

## What was wrong

`TeeRating` validated the ranges of an 18-hole course:

```python
MIN_COURSE_RATING = 55   MAX_COURSE_RATING = 85
MIN_SLOPE_RATING  = 55   MAX_SLOPE_RATING  = 155
MIN_PAR           = 66   MAX_PAR           = 76
```

The `Tee` entity accepts CR 45–90 and SR 40–160, so we store courses we then refuse to rate. In the versioned RFEG dataset (4304 tees):

| | Real range | `TeeRating` accepted |
|---|---|---|
| Course rating | 46.5 – 84.7 | 55 – 85 |
| Slope rating | 46 – 157 | 55 – 155 |
| Par | 54 – 74 | 66 – 76 |

**468 tees across 227 courses** fall outside on CR, **642 tees** on par.

Two consequences:

**Quick match — silent and wrong.** The tee is skipped and its players play off the raw Handicap Index. On Son Parc's orange tee (par 58, CR 54.9, SR 91) an 18 handicap receives **18 strokes where the WHS gives 11**. Off by the whole handicap, on a short course where that decides everything.

**Competition — a 500.** `TeeContextBuilder.build` rates *every* tee of the course, so one short tee anywhere makes the round impossible to generate. Verified with all four players enrolled on YELLOW, a perfectly valid tee: still 500, because ORANGE exists in the course. Those 227 courses cannot host a competition at all.

## What changed

**The ranges are the union of every course type** — the same ones `Tee` already validates. The strict per-type range stays in `GolfCourse._validate_tee_ratings`, which is the only place that knows the `course_type`:

| | Standard 18 | Executive | Pitch & putt |
|---|---|---|---|
| Slope | 55 – 160 | 40 – 155 | 40 – 155 |
| Rating | 50 – 90 | 45 – 90 | 45 – 90 |

I first considered moving those per-type ranges into `TeeRating`, but this domain cannot import `golf_course` without breaking a separation the code defends explicitly. Keeping the wide range here and the strict one there means the net against a mistyped CR on a standard 18 is not lost — it just lives where the course type is known.

**`_rating_for` no longer raises from its own fallback.** The retry reused the same course rating and slope rating, changing only the par, so a rating out of range raised twice and the ValueError reached the API. It now returns `None`, the tee is left out of the context, and its players fall back to the Handicap Index — which is what the docstring promised. No tee in the current catalogue takes that path; it is a backstop.

**The parity harness covers a short course.** The client derives par from the holes while the backend carries it in the `TeeRating`, so this is exactly where the two implementations can drift. `run` now takes the scenario's own pars and stroke index instead of always emitting Meis'.

## Testing

`pytest tests/unit/ -q`: 2912 passing. ruff and mypy clean (the 6 ruff findings in `main.py` and `test_quick_match_endpoints.py` are pre-existing on `develop`).

New tests: a federated pitch & putt is now rated; the extremes of the catalogue (CR 46.5–84.7, SR 46–157) all fit; and an unratable tee is left out of the context instead of raising. Existing range tests updated to the new bounds.

Regenerating `backendParity.json` is **left to the RyderCupWeb PR**, which is where it is consumed and where `RATED_*` moves in step. Running the generator today already reports the gap that PR will close:

```
AssertionError: hándicap de juego ... en "campo corto": expected 18 to be 11
```

None of the 12 existing parity scenarios changed value.

## Deploying

**Backend first, then the frontend, back to back.** Between the two deploys, quick match on those 227 courses will allocate differently online and offline.

One more thing worth stating: competition playing handicaps are frozen into `matches` when the round is generated, so nothing already played moves. Quick match recomputes on every read, so **a quick match in progress on one of these courses will change allocation mid-round** — from 18 strokes to 11 on the example above. Correct, but visible to someone who did not touch anything.

Closes #206
Closes #219

## Summary by CodeRabbit

- **New Features**
  - Added support for pitch-and-putt courses and short 18-hole courses with broader par, course-rating, and slope-rating ranges.
  - Unratable tees are now excluded from tee ratings while hole-card information remains available.

- **Bug Fixes**
  - Improved handling of invalid tee and course ratings without interrupting course setup.
  - Preserved fallback behavior when valid course-level ratings are available.

- **Tests**
  - Added coverage for pitch-and-putt ratings, unratable tees, and rating boundary values.

### CODE REVIEW

**Comment by agustinEDev** [comment]
Review of the branch turned up seven findings. Two are fixed here, one is a product decision now taken, and the rest are recorded.

## Fixed

**The comment lied about what an unrated tee does** (`ec06770`). It said the tee's players would fall back to their Handicap Index. They do in quick match; they do not here — `generate_matches` and `reassign_match_players` treat a missing rating as `TeeColorNotFoundError`, so a player enrolled on an unratable tee still cannot have their round generated. What this PR changes for them is a 500 into a 400 with a message. Carrying the real fallback into competition touches the five points that raise today and is a separate change. The docstring now says exactly that.

**The common fallback went silent** (`ec06770`). Replacing the `logger.debug` with a bare `pass` muted the branch that fires when a tee's own par is unusable — the *common* one, since 25 of the 800 imported courses change par between tees — and taking it silently moves that player's course handicap. The log is back.

Plus two blank lines before the new test class (`2c64a90`).

## Decided, not overlooked

**Finished quick matches get re-scored on deploy.** Quick-match strokes and results are not persisted — `get_recent_matches_use_case` says it outright: "El resultado no está guardado en ningún sitio: se recalcula". So this does not only affect matches in progress: every match already played on one of these 227 courses is re-scored with the new allocation the moment this ships, and one that finished 1 up can show in the history as halved.

Decision taken deliberately: production has had a couple of users so far, so getting the allocation right is worth more than protecting history that was computed wrong in the first place. Recording it here so it is a decision and not a surprise.

Competition is unaffected — its playing handicaps are frozen into `matches` when the round is generated.

## Recorded, not changed

**An unratable tee silently loses its gender.** `generate_matches_use_case.py:561` does `if tee_key not in tee_ratings: tee_key = (color, None); tee_gender = None`. That path was unreachable before — `build` either rated every tee or raised — and this PR makes it reachable. In SCRATCH mode nothing checks the rating afterwards, so the `MatchPlayer` persists with `tee_gender=None` and later per-tee card lookups resolve to the course card instead of the player's. Same shape in `reassign_match_players_use_case.py:239`. No tee in the current catalogue reaches it, so it stays a backstop, but it is worth its own issue rather than a quiet fix inside this one.

**The parity harness cannot protect this release on its own.** The frontend still holds the old `RATED_*` bounds, so with the backend deployed alone the same match allocates differently online and offline — verified: PH 11 and 21 on the backend against 18 and 30 on the frontend for the short-course scenario, 10 strokes of difference against 12. And `backendParity.json` has not been regenerated, so the FE parity test stays green over 12 scenarios that contain no short course. Both close together in the RyderCupWeb PR, which regenerates the fixture and moves the constants. Until then this must not deploy alone.

**Comment by agustinEDev** [comment]
@coderabbitai full review

File: src/modules/competition/application/services/tee_context_builder.py [hunk 1]
```diff
@@ -70,7 +70,12 @@ def build(golf_course: GolfCourse) -> TeeContext:
         for tee in golf_course.tees:
             gender = tee.gender.value if tee.gender else None
             tee_key = (tee.color.value, gender)
-            tee_ratings[tee_key] = TeeContextBuilder._rating_for(tee, course_par, golf_course)
+            # La barra que no se puede valorar se queda FUERA del diccionario en
+            # vez de tumbar la construccion entera: quien la juegue jugara con su
+            # Handicap Index, y el resto del campo sigue funcionando. Ver #219.
+            rating = TeeContextBuilder._rating_for(tee, course_par, golf_course)
+            if rating is not None:
+                tee_ratings[tee_key] = rating
```
[the remark this answers is not in the document]
Comment (agustinEDev) [reply src/modules/competition/application/services/tee_context_builder.py:78]: Fixed in 4e4221f. Both messages were wrong, and the second one embarrassingly so.

The debug line claimed an unusable par on every failure, though the `except` catches an out-of-range course rating or slope rating just as well — and it was emitted *before* the retry, so it announced a fallback that could fail one line later. It now fires only when the retry succeeds and names both pars. The retry itself is skipped when the tee has no card of its own: with the same par the second attempt is the first one again, and what failed was the rating, not the par.

The warning still promised the Handicap Index fallback. That is the exact sentence I corrected in the docstring one commit earlier and left standing ten lines below it — same defect, same function. It now says what competition actually does: the tee is left out and the round cannot be generated for whoever is enrolled on it, with the CR, SR and par that made it unratable.

Added a test that pins the wording, including that "Handicap Index" must *not* appear in it.
